mod/events.php

```diff
@@ -150,26 +150,16 @@ function events_post(App $a)
 		DI::baseUrl()->redirect($onerror_path);
 	}
 
-	$share = intval($_POST['share'] ?? 0);
-
-	$c = q("SELECT `id` FROM `contact` WHERE `uid` = %d AND `self` LIMIT 1",
-		intval(local_user())
-	);
-
-	if (DBA::isResult($c)) {
-		$self = $c[0]['id'];
-	} else {
-		$self = 0;
-	}
+	$self = \Friendica\Model\Contact::getPublicIdByUserId($uid);
```

Comment:
The function `getPublicIdByUserId` does return the public contact id of the self contact. The result is not the same as the result of the replaced query.

Reply:
This query is used to populate `$self` which is only used to generate self-only permission string (in lines 176 and 179) which should use the public contact id of the local user.

Reply:
AFAIK the ACL always worked with the user's contact ids. This is something that I have to check against before being able to merge it.

Reply:
We use public contact ids for the rest of the permissions, I believe this is a good normalization change.

Reply:
Looking at https://github.com/friendica/friendica/blob/2020.12-rc/src/Repository/PermissionSet.php#L163-L182 it seems that we expect the user's contact ids. And looking at the way the function are called: https://github.com/friendica/friendica/blob/2020.12-rc/src/Protocol/ActivityPub/Transmitter.php#L248-L250 or https://github.com/friendica/friendica/blob/2020.12-rc/src/Module/Profile/Status.php#L219-L221 or https://github.com/friendica/friendica/blob/2020.12-rc/src/Module/Objects.php#L83-L85 it seems as if we are expected to use the user's contact id.

Reply:
The public contact id or the self contact id?

Reply:
AFAIK (when looking at the links) we always provide the user's contact id to the ACL fetching when reading posts. So I guess we should only feed the ACL with the user's contact id as well.

Reply:
Again, the public contact id or the self contact id? Both can be described as the "user's contact id".

Reply:
The public id always belongs to the "uid=0", the self contact always belongs to the user, so the self contact is the user's contact id.

I just created a PR to handle this: https://github.com/friendica/friendica/pull/9732

Reply:
Thank you for the elaboration, looking forward to the end of this sort of confusion.